packages/next/next-server/lib/post-process.ts

```diff
@@ -149,6 +152,83 @@ class FontOptimizerMiddleware implements PostProcessMiddleware {
   }
 }
 
+class ImageOptimizerMiddleware implements PostProcessMiddleware {
+  inspect(originalDom: HTMLElement, _data: postProcessData) {
+    const imgElements = originalDom.querySelectorAll('img')
+    let eligibleImages: Array<HTMLElement> = []
+    for (let i = 0; i < imgElements.length; i++) {
+      if (isImgEligible(imgElements[i])) {
+        eligibleImages.push(imgElements[i])
+      }
+      if (eligibleImages.length >= MAXIMUM_IMAGE_PRELOADS) {
```

Comment:
5 by 500 would (just) not qualify for preload since the logic down at line 207 checks to see if width * height is less than or equal to 2500 and disqualifies if that's so. That's pretty arbitrary, though--do you think a different cutoff would be better?

Reply:
oh sorry I mis-read this line. This makes sense. I feel its ok to start with arbitrary number and change it with some experiment results